components/misc/rng.cpp

```diff
@@ -5,27 +5,33 @@
 
 namespace Misc
 {
+    Rng Rng::sInstance{};
```

Comment:
Probably there are ways to simplify random generator. Now it looks fishy and provides two sets of API - one set for your purpose and one for static random generator instance, which is used in the rest of the engine.

Reply:
Maybe something like this?

Reply:
Yes, something like this.